pkgs/top-level/all-packages.nix

```diff
@@ -5931,7 +5931,9 @@ with pkgs;
 
   self = callPackage_i686 ../development/interpreters/self { };
 
-  spark = callPackage ../applications/networking/cluster/spark { };
+  spark = spark_21;
+  spark_16 = callPackage ../applications/networking/cluster/spark { version = "1.6.0"; };
```

Comment:
This should be probably updated to the latest version of the 1.x series (1.6.3).